source/simulator/solver_schemes.cc

```diff
@@ -1069,6 +1070,124 @@ namespace aspect
     if (parameters.run_postprocessors_on_nonlinear_iterations)
       postprocess ();
   }
+
+
+
+  template <int dim>
+  void Simulator<dim>::solve_stokes_adjoint ()
+  {
+    // TODO iterate
+    // todo make number of interations a parameter and change Do iterations for inversion parameter
+//    for (unsigned int i=0; i<parameters.num_it_adjoint; ++i)
+//      {
```

Comment:
can you just use the `max_nonlinear_iterations` parameter that is already around? How would such a nonlinear scheme ook like? Would you just repeat the whole process (forward + adjoint) n times? Or do you need different number of iterations for the forward problem and the overall problem? I

Reply:
Maybe I could use that parameter. The scheme would just repeat the whole process n times. The aspect I'm still trying to figure out how to do best is to save the input from several iterations. At the moment the iteration is done in core.cc (for (unsigned int i=0; i<=parameters.num_it_adjoint; ++i) and then the iterations are saved as different timesteps but this iteration should of course happen here. Do you have any pointers for where to look regarding saving iterative output?